src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -369,8 +373,18 @@ private static bool TryParseEnumCore(ref Utf8JsonReader reader, JsonSerializerOp
             int charsWritten = reader.CopyString(charBuffer);
             ReadOnlySpan<char> source = charBuffer.Slice(0, charsWritten);
 
-            // Try parsing case sensitive first
-            bool success = Enum.TryParse(source, out T result) || Enum.TryParse(source, ignoreCase: true, out result);
+            bool success;
+            T result;
+            if ((_converterOptions & EnumConverterOptions.AllowNumbers) != 0 || !Regex.IsMatch(source, NumericPattern))
```

Comment:
Per the build failures, could you try using the regex generator here?

Reply:
Fixed.